src/opover.d

```diff
@@ -1959,14 +1961,31 @@ extern (C++) static Dsymbol inferApplyArgTypesX(Expression ethis, FuncDeclaratio
         }
         if (!inferApplyArgTypesY(tf, parameters, 1))
             m = MATCHnomatch;
+
         if (m > match)
         {
             fd_best = f;
             fd_ambig = null;
             match = m;
         }
-        else if (m == match)
+        else if (m == match && fd_best)
+        {
+            auto tf1 = cast(TypeFunction)fd_best.type;
+            auto tf2 = cast(TypeFunction)f      .type;
+            tf1 = cast(TypeFunction)Parameter.getNth(tf1.parameters, 0).type.nextOf();
```

Comment:
This separation is just to avoid writing too-long a line:

```d
auto tf1 = cast(TypeFunction)Parameter.getNth((cast(TypeFunction)fd_best.type).parameters, 0).type.nextOf();
```

The first set value is merely an intermediate result, so naming it is just useless.